### PR TITLE
Feature/convert to json

### PR DESCRIPTION
**Pull request by davidrajones**

### CODE REVIEW

File: perl/bin/alleleCounterToJson.pl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  run($options);
+}
+
+sub run {
+  my ($options) = @_;
+  my $json_string = Sanger::CGP::AlleleCount::ToJson::alleleCountToJson($options->{'a'}, $options->{'l'});
+  my $OUT;
+  open($OUT, '>', $options->{'o'}) or croak("Error opening file for output: $!");
+    print $OUT "$json_string";
+  close($OUT) or croak("Error closing output file for JSON conversion: $!");
```
Comment (keiranmraine) [inline perl/bin/alleleCounterToJson.pl:47]: Does this work without throwing error about attempting to open/close stdout?

I thought you needed to do:

```perl
my $OUT = *STDOUT;
if(defined $options->{'o'}) {
    open($OUT, '>', $options->{'o'}) or croak("Error opening file for output: $!");
}

print $OUT "$json_string";

if(defined $options->{'o'}) {
    close($OUT) or croak("Error closing output file for JSON conversion: $!");
}
```

File: perl/bin/alleleCounterToJson.pl [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+sub option_builder {
+  my ($factory) = @_;
+
+  my %opts;
+
+  &GetOptions (
+        'h|help'    => \$opts{'h'},
+        'l|locus-file=s' => \$opts{'l'},
+        'a|allelecount-file=s' => \$opts{'a'},
+        'o|output-file=s' => \$opts{'o'},
```
Comment (keiranmraine) [inline perl/bin/alleleCounterToJson.pl:60]: `=s` means the option is required, while `:s` means optional.  I'm not convinced it makes that much difference more an indicator of intent.

**Review by keiranmraine — COMMENTED** [review]
Couple of things to check over, otherwise it's good

File: perl/bin/alleleCounterToJson.pl (hunk 2 above)
Comment (davidrajones) [reply perl/bin/alleleCounterToJson.pl]: Corrected

File: perl/bin/alleleCounterToJson.pl [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+use Getopt::Long 'GetOptions';
+use Pod::Usage;
+
+use Sanger::CGP::AlleleCount;
+use Sanger::CGP::AlleleCount::ToJson;
+
+{
+  my $options = option_builder();
+  $options->{'o'} = '/dev/stdout' unless(defined $options->{'o'});
```
Comment (keiranmraine) [inline perl/bin/alleleCounterToJson.pl:37]: This isn't needed now you've changed the other bit

File: perl/bin/alleleCounterToJson.pl (hunk 1 above)
Comment (davidrajones) [reply perl/bin/alleleCounterToJson.pl]: No errors thrown in testing, but added the catch in case.

**Pull request merged**